recon/src/ahash.rs

```diff
@@ -284,8 +293,8 @@ mod tests {
     fn push_add() {
         let plus = Sha256a::digest("hello") + Sha256a::digest("world");
         let mut push = Sha256a::identity();
-        push.push((&"hello".to_string(), &Sha256a::digest("hello")));
-        push.push((&"world".to_string(), &Sha256a::digest("world")));
+        push.push((&"hello".as_bytes().into(), &Sha256a::digest("hello")));
+        push.push((&"world".as_bytes().into(), &Sha256a::digest("world")));
```

Comment:
FYI rust has the `b"world"` syntax which will treat a string literal as a byte array instead of a string. No need to change here.

Reply:
that gave me an error

```
error[E0277]: the trait bound `&eventid::EventId: From<&[u8; 5]>` is not satisfied
   --> recon/src/ahash.rs:296:20
    |
296 |         push.push((b"hello".into(), &Sha256a::digest("hello")));
    |                    ^^^^^^^^ ---- required by a bound introduced by this call
    |                    |
    |                    the trait `From<&[u8; 5]>` is not implemented for `&eventid::EventId`
    |
    = help: the following other types implement trait `From<T>`:
              <eventid::EventId as From<&Vec<u8>>>
              <eventid::EventId as From<&[u8]>>
    = note: required for `&[u8; 5]` to implement `Into<&eventid::EventId>`
```

`From<&[u8; 5]>` != `<eventid::EventId as From<&[u8]>>`
    
Am I missing something?

Reply:
You still need the reference `&`, I would expect this to work `push.push((&b"hello".into(), &Sha256a::digest("hello")));` althought you might need to play with operator precedence `push.push((&(b"hello".into()), &Sha256a::digest("hello")));`